src/main/kotlin/graphs/SimpleGraph.kt

```diff
@@ -218,6 +218,22 @@ open class SimpleGraph<V> {
         }
     }
 
+    /**
+     * Checks if [other] is an exact copy of this graph. This means that [other] needs to contain the same
+     * vertex objects as this graph and the edges of [other] need to be between the same vertices as the edges of
+     * this graph.
+     *
+     * @return True if [other] is an exact copy of this graph. False otherwise.
+     */
+    override fun equals(other: Any?): Boolean {
+        return other != null &&
+            other is SimpleGraph<*> &&
+            other.n == this.n &&
+            other.m == this.m &&
+            other.V.all { it in this.V } &&
+            other.map == this.map
```

Comment:
So wie ich das verstehe, nutzt _map_ die equals() Methoden der Typen, die in der Map drin sind.
Vielleicht sollte man in den Docs anmerken, dass man einen Typen verwenden muss, der das korrekt umsetzt?